### ISSUE TITLE
Shampoo Optimizer

### ISSUE THREAD

**Issue by ClashLuke** [HomebrewNLP/HomebrewNLP-Jax#15]
Second-order optimizers such as [K-Fac](https://arxiv.org/abs/1503.05671), [LBFGS](https://pytorch.org/docs/stable/generated/torch.optim.LBFGS.html) and [AdaHessian](https://arxiv.org/abs/2006.00719) promise significantly improved convergence rates at horrific memory costs. [Scalable Shampoo](https://arxiv.org/abs/2002.09018) promises a low memory footprint and vectorisable computation while retaining the convergence advantage of other second-order optimisers. Adding it to our code could reduce training time by 10% or even up to an order of magnitude.\
This issue is about implementing shampoo ([reference](https://github.com/google-research/google-research/tree/master/scalable_shampoo/optax) might help), running a hyperparameter sweep to find its best configuration and comparing the best possible runtime with our previous best.
